### PR TITLE
fix(ci): enable Release Action for protected branches

### PR DESCRIPTION
**Pull request by jenstroeger**
Thanks you @lecram for looking into this!

### CODE REVIEW

File: .github/workflows/check-and-release-main.yaml [hunk 1]
```diff
@@ -51,9 +51,11 @@ jobs:
       uses: step-security/harden-runner@248ae51c2e8cc9622ecf50685c8bf7150c6e8813
       with:
         egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
-
-    - uses: actions/checkout@d0651293c4a5a52e711f25b41b05b2212f385d28
+    - name: Checkout
+      uses: actions/checkout@d0651293c4a5a52e711f25b41b05b2212f385d28
       with:
+        # For protected branches, see: https://github.com/relekang/python-semantic-release/issues/311
+        token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (behnazh) [inline .github/workflows/check-and-release-main.yaml:58]: Is this really necessary for this repo? The `token` secret should be available by default in `checkout@v3` according to the documentation [here](https://github.com/actions/checkout#push-a-commit-using-the-built-in-token).

Comment (jenstroeger) [reply .github/workflows/check-and-release-main.yaml:58]: I added this for documentation’s sake with a reference to a related issue. Private repos with protected branches may need to set that token to make the Action work.

Comment (behnazh) [reply .github/workflows/check-and-release-main.yaml:58]: > I added this for documentation’s sake with a reference to a related issue. Private repos with protected branches may need to set that token to make the Action work.

How about adding this to the comment?

Comment (jenstroeger) [reply .github/workflows/check-and-release-main.yaml:59]: Commit https://github.com/jenstroeger/python-package-template/pull/209/commits/0fd24a03caaa718786a718220bc8ca0319ea0495.

**Review by behnazh — APPROVED** [review]

**Pull request merged**